src/VivViewer.js

```diff
@@ -2,43 +2,155 @@ import React, { PureComponent } from 'react';
-    // https://github.com/hubmapconsortium/vitessce-image-viewer/issues/107
-    return loader.isPyramid
-      ? new VivViewerLayer({
-          id: `VivViewerLayer-${loader.type}`,
+    const { loader, overviewOn } = this.props;
+    if (loader.isPyramid) {
+      const layers = [
+        new VivViewerLayer({
+          id: `${loader.type}-detail`,
+          // Because TileLayer is unique in updating on viewport changes,
+          // it needs to be aware of what viewport it is rendering in -
+          // layerFilter only handles `draw` calls.
+          viewportId: 'detail',
           ...this.props
         })
-      : new StaticImageLayer({
-          id: `StaticImageLayer-${loader.type}`,
-          ...this.props
+      ];
+      if (overviewOn) {
+        const { viewHeight, viewWidth } = this.props;
+        const { viewState } = this.state;
+        const viewport = new OrthographicView().makeViewport({
+          // From the current `detail` viewState, we need its projection matrix (actually the inverse).
+          viewState: viewState.detail,
+          height: viewHeight,
+          width: viewWidth
         });
+        // Use the inverse of the projection matrix to map screen to the view space.
+        const boundingBox = [
+          viewport.unproject([0, 0]),
+          viewport.unproject([viewport.width, 0]),
+          viewport.unproject([viewport.width, viewport.height]),
+          viewport.unproject([0, viewport.height])
+        ];
+        layers.push(
+          new OverviewLayer(this.props, {
+            id: `${loader.type}-overview`,
+            boundingBox,
+            ...this.props
+          })
+        );
+        return layers;
+      }
+      return layers;
+    }
+
+    return new StaticImageLayer({
+      id: `StaticImageLayer-${loader.type}-detail`,
+      ...this.props
+    });
   }
 
   render() {
     /* eslint-disable react/destructuring-assignment */
+    const { loader, overviewOn } = this.props;
+    const { numLevels } = loader;
+    const { imageWidth, imageHeight } = loader.getRasterSize({
+      z: 0
+    });
     const views = [
       new OrthographicView({
-        id: 'ortho',
+        id: 'detail',
         controller: true,
         height: this.props.viewHeight,
         width: this.props.viewWidth
       })
     ];
-    const { initialViewState } = this.props;
+    if (loader.isPyramid && overviewOn) {
+      /* eslint-disable no-bitwise */
+      views.push(
+        new OrthographicView({
+          id: 'overview',
+          controller: false,
+          // There's probably a better way to do this, but I think there needs to be
+          // some investigation into padding - look at the tiff vs zarr maps.
+          height: imageHeight >> (numLevels - 1),
+          width: imageWidth >> (numLevels - 1),
+          x: this.props.viewWidth - (imageWidth >> (numLevels - 1)) - 25,
+          y: this.props.viewHeight - (imageHeight >> (numLevels - 1)) - 25,
```

Comment:
not sure what this is doing? what's the -25?

Reply:
An offset from the corner.  This reminds me that Nils asked for the ability to place this in different locations.  I'll update and add a comment.

Reply:
yeah I'd also avoid bitwise here. It's relying on both the data being downscaled by a factor of two as well as the size of the pyramid. Feels like we could come up with a solution that deals with scaling the overview image relative to the viewport width or something